### DISCUSSION TITLE
Some newbie (sensor) questions

### DISCUSSION THREAD

**Discussion by DamijanD** [gekkekoe/esphome-ecodan-hp#71] [Announcements]
Hi.

As a proud owner of new atom (with 2 zones + melcloud) I first need to say big thanks to @gekkekoe for implementation and @fonske for HW delivery.

And now some sensor questions:
- `3-way Valve` and `3-way Valve 2`: shouldn't this be one sensor with state: position1(heating) or position2(DHW) at least this is how I see it - so this is the same 3-way valve in different positions
- What is the difference between `Estimated Output Power` ( 4.737 kW) and `Output Power` (4.000 kW) I have noticed that later is always floor of the first one?
- `Water Pump Status` this is internal pump, and `Water Pump 2 Status` is external pump. I am missing the second external water pump - each zone has one. Via melcloud I can get `IdleZone1` and `IdleZone2`. And this is the only thing that I am missing in this device/implementation.
- In HomeAssistant by default I get 2 entities for each zone:`ecodan_heatpump_zone_1_temperature` and `ecodan_heatpump_zone_1_climate` what is the difference between the two?

And some newbie observations/questions:
- if you want static IP, that needs to be set in firmware
- when you first start it, it could mess up your settings (DWH was set to 50° and room temp was set to 16°)
- as this is my first EPShome device there are some other questions:
1. How can I upgrade firmware via web interface. On http://[IP of the device] there is no upgrade option
2. ESPHome integration for HA: will it detect device by itself or do I need to add it manually? If I add it will it add new device to HA as well (I don't want a duplicate to be created)?

**Comment by F1p** [comment]
**3-way Valve/3-way Valve 2:** 
Some models of package have more than one, usually 3-way valve 1 is for DHW/Heating where valve 2 might be for boiler & heat pump combo system

**What is the difference between Estimated Output Power ( 4.737 kW) and Output Power (4.000 kW)**
Estimated Output Power is based on a Formula the ESP is performing, "Output Power" is provided by the FTC and is rounded to floor 1kW - is it the same with input power (don't think input implemented in this repo)

**Water Pump Status this is internal pump, and Water Pump 2 Status is external pump. I am missing the second external water pump - each zone has one.** 
Pump 1 = Primary Water, Pump 2 = Zone 1 (not connected though), Pump 3 = Zone 2, Pump 4 = Hot Water, Pump 13 = Zone 1 for me

I don't think all the pumps were implemented in this repo, you could make a pull/feature request


**Via melcloud I can get IdleZone1 and IdleZone2. And this is the only thing that I am missing in this device/implementation.**
I don't think Gekkekoe has a 2 zone system so didn't implement, you could make a pull/feature request

**Comment by gekkekoe** [comment]
thx @F1p for answering most questions. 

- the climates are the entities to control/set temps. 
- see esphome for static ip config examples. You can also bind a ip to your esp, the. it's kinda static.
-  updates can be done via cmd line. see main page of the repo. It can also be done via HA. If you want update via the web, recompile using Arduino lib (I would advise against this)
- HA will auto detect, but you can add it manually (when. you deleted the entity)
- for missing sensors, make a PR or request it. I am a bit busy lately.

**Reply by DamijanD** [reply]
Just for information: this implementation now has much better zone2 support as it did 2 weeks ago. There is only one small (more cosmetic issue): As you can see in image the climate/temperature control reports Heating for both zones even though only Z2 is running (and vice versa)
![image](https://github.com/user-attachments/assets/cf586e3e-f086-4c8e-b11e-bda589ae69ee)

But I didn't find where in the code this is set. Probably more accurate approach would be "whatever is used now" && pump status or directly new `MultiZoneStatus`

**Reply by gekkekoe** [reply]
It's a bit more complicated, the mode is derived from the current operation. The climate component are the same one used for regular temp/flow and dhw. So probably need some indicator function needs to be passed as arg to the climate. It needs some more thought. (It also needs to work for folks without 2 zones, then the multi zone indicator always displays idle)

**Reply by F1p** [reply]
@DamijanD @gekkekoe 
This was my logic for implementing thermostat running settings per zone:
[zone1 logic](https://github.com/F1p/Mitsubishi-CN105-Protocol-Decode/blob/8cf6235c45c7a496afdfca09aac3a1e795aa189f/MQTTDiscovery.h#L558 )
[zone2 logic](https://github.com/F1p/Mitsubishi-CN105-Protocol-Decode/blob/8cf6235c45c7a496afdfca09aac3a1e795aa189f/MQTTDiscovery.h#L559 )

**Reply by gekkekoe** [reply]
So have been giving this some thoughts.
we could probably even detect the running state by determining setpoint < current temp right ? Are there scenario where a zone is not running under those conditions ?

**Reply by merlin051** [reply]
Just chipping in - I think a buffer tank would be an exception to this rule

**Reply by gekkekoe** [reply]
> Just chipping in - I think a buffer tank would be an exception to this rule

Ah good point, then need to derive it from the multi status then.

**Reply by F1p** [reply]
> So have been giving this some thoughts. we could probably even detect the running state by determining setpoint < current temp right ? Are there scenario where a zone is not running under those conditions ?

And in AA it can be running at +1 over setpoint too

**Reply by DamijanD** [reply]
> > So have been giving this some thoughts. we could probably even detect the running state by determining setpoint < current temp right ? Are there scenario where a zone is not running under those conditions ?
> 
> And in AA it can be running at +1 over setpoint too

And only one Zone can have AA - so other zone usually uses WC which does not require to have room temperature sensor at all, but if you have it, it is actually almost not used. From my experience it never stops heating if above setpoint. But if it is in idle state (because DWH or something else) then it won't start heating until current temp is lower or equal set point.

**Comment by gekkekoe** [comment]
@DamijanD  and others with multi zone systems, please give https://github.com/gekkekoe/esphome-ecodan-hp/tree/adjust_climates_for_multizones a spin and report back if its working correctly. 

Be sure to enable zone2.yaml, that triggers a  required substitution

**Reply by DamijanD** [reply]
After few days I don't see anything different. Today was first day that we had situation when only Z2 was working, but I didn't see any difference with behaviour:
![image](https://github.com/user-attachments/assets/8abbc52f-db7f-4d09-9678-5cc23b44078f)

**Reply by gekkekoe** [reply]
hmm weird, so the multi zone byte would have been 3 (z2 working).
When you include yaml, it should have set the identifier as follows:
zone1: 2
zone2: 3

Zone 1 can only be active only if one of the following conditions apply:
https://github.com/gekkekoe/esphome-ecodan-hp/blob/9c57780f25c2dee12e2132b6d68193f58eff9326/components/ecodan/climate.cpp#L53

```
                        if (this->climate_zone_identifier == ClimateZoneIdentifier::SINGLE_ZONE
                            || static_cast<uint8_t>(ClimateZoneIdentifier::MULTI_ZONE_BOTH) == status.MultiZoneStatus
                            || static_cast<uint8_t>(this->climate_zone_identifier) == status.MultiZoneStatus) {
```
For zone 1, the zone identifier should have been 2, so first condition is false
It's also not 1 for both. 

So I'm having the impression that you are not running the latest code or the subst went wrong.
Can you check in `.esphome/build/src/main.cpp` and look for `heatpump_climate_z1->set_zone_identifier` and see what parameter number is supplied there ? It should be 2. If its missing then you are running main branch.

You might need to do a clean before compile. Also it can take up to 2 mins before all status are published, so we could have a small lagging window.

**Reply by DamijanD** [reply]
code looks ok:
![image](https://github.com/user-attachments/assets/622c5c55-c4ae-424a-8ea3-a70247d5bffd)
![image](https://github.com/user-attachments/assets/7e51d27e-b0b3-402c-a0a3-643201db44be)
Is it possible that we are missing part where we set it back to idle? If maybe after DHW cycle we get brief "Both" and then we get from "Both" to Z1 or Z2 ? Even though it does not look like it:
![image](https://github.com/user-attachments/assets/1fe6c578-78ce-42c0-9c09-5e60896b9c94)

**Reply by gekkekoe** [reply]
hmm weird, but we know that the status was 3. (seen via translation, this value is instant, no delay).
Could be small window where it turns to idle, but idle means off. So that does not compute. The set back to idle is done by the next get multi zone status. But the climates only see the 'current' state, so that's not possible as well.

Only cause I can think of is when you change a climate setting, it has a no-update time before it accepts new states (60s). So if you change (set new target temp) the climate within 60s ago, this might have happen. But this is all guessing.

Do you have logging with time for the multibyte byte and climate for zone 1?

**Comment by gekkekoe** [comment]
I think I've found it, had the modification on mode instead of action....
Please rebuild and retest (just pushed a fix on the branch)

**Reply by DamijanD** [reply]
Now it works perfectly: 
![image](https://github.com/user-attachments/assets/6016cd80-90ab-4afd-ae40-b38dcc28e78a)
![image](https://github.com/user-attachments/assets/a07cc42e-8152-46bc-94af-a6a776c718b6)
![image](https://github.com/user-attachments/assets/4e87cde0-8483-43c1-9592-5e8cd80ce7c6)
Thx.

**Reply by gekkekoe** [reply]
merged thnx for confirming

**Reply by F1p** [reply]
Maybe not strictly true because your showing mode heating/cooling, i like to show Idle = Off because visually i know that zone is not working and same with DHW climate

![image](https://github.com/user-attachments/assets/20649b3c-2e99-41ef-898f-90f9d14d207d)

**Reply by DamijanD** [reply]
IMHO: If you check the main controller on HP
**pause** symbol should match Idle (and when heating is on but zones don't need actual heating there is pause symbol)
**stop** symbol should match Off

**Reply by gekkekoe** [reply]
I prefer idle, because the controls are still available (you still can set a new setpoint).

**Reply by F1p** [reply]
> I prefer idle, because the controls are still available (you still can set a new setpoint).

suggestion Off = Prohibit active?

**Comment by F1p** [comment]
@DamijanD are you able to check status of message 0x28, byte 11 please?
It is observed on startup that MELCloud writes message 0x34 byte 11 and i think this 0x28 byte turns on because of it

**Reply by DamijanD** [reply]
Yes I can observe that byte.
Just general value of 0x28[11] or do I need to do something with melcloud as well (some action which should cause 0x34 message)?

**Reply by F1p** [reply]
> Yes I can observe that byte. Just general value of 0x28[11] or do I need to do something with melcloud as well (some action which should cause 0x34 message)?

Observe before, then unplug MELCloud and plug back in, observe after please

**Reply by DamijanD** [reply]
byte 11 is 0 all the time, but after unplug/plug it was 1 (probably only for one reading as we read every ~21s and the duration of that state was for 23s)
default: RES { .Hdr { FC, 62, 02, 7A, 10 } .Payload { 28, 00, 00, 00, 00, 00, 00, 01, 00, 01, 00, **00**, 00, 00, 00, 00 } .Chk { E8 } }
after pl: RES { .Hdr { FC, 62, 02, 7A, 10 } .Payload { 28, 00, 00, 00, 00, 01, 00, 01, 00, 01, 00, **01**, 00, 00, 00, 00 } .Chk { E6 } }

Hope it helps...

**Reply by F1p** [reply]
That is my observed behaviour - it seems the MEL is writing 0x34 byte 11 on connection, which triggers 0x28 byte 11 for one cycle, like you saw
We don't know why/what this is though

**Reply by gekkekoe** [reply]
so we found another set cmd, now see if we can find what it sets...
